public/js/spreadsheet.js

```diff
@@ -1,26 +1,48 @@
 document.addEventListener( 'DOMContentLoaded', () => {
+  const debounce = (func, wait = 100) => {
+    let timeout;
+    return function(...args) {
+      clearTimeout(timeout);
+      timeout = setTimeout(() => {
+        func.apply(this, args);
+      }, wait);
+    };
+  };
+
   let spreadsheetBlocks = document.querySelectorAll( '.block-spreadsheet' );
 
   [...spreadsheetBlocks].forEach( block => {
 
     // Filter Spreadsheet table by entered value in search field
-    let searchInput = block.querySelector( '.spreadsheet-search' );
-    let table = block.querySelector( '.spreadsheet-table' );
-    let rows = [...table.querySelectorAll( 'tr:not(:first-child)' )];
-    let headerCells = [...table.querySelectorAll( 'tr:first-child th' )];
-    let defaultSortButton = block.querySelector( '.spreadsheet-default-sort' );
-    let emptyMessage = block.querySelector( '.spreadsheet-empty-message' );
+    const searchInput = block.querySelector( '.spreadsheet-search' );
+    const table = block.querySelector( '.spreadsheet-table' );
+    let tableBody = table.querySelector( 'tbody' );
+    let rows = [...tableBody.querySelectorAll('tr')];
+    const headerCells = [...table.querySelectorAll( 'thead th' )];
+    const emptyMessage = block.querySelector( '.spreadsheet-empty-message' );
+    let previousSearchTerm;
 
-    searchInput.addEventListener( 'input', () => {
-      let searchTerm = searchInput.value;
+    const filterRows = ( event ) => {
       let hasMatches = false;
+
+      const searchTerm = event.target.value.toLowerCase().trim();
+
+      previousSearchTerm = searchTerm;
```

Comment:
It looks like this is not used anywhere, right? I didn't notice at first, ESLint complained about the definition in line 23 not being used.

Reply:
Yes, this was a leftover from a previous approach, removed it now